### PR TITLE
Draft pass to update the InputSystem to use the "Logical" InputAction

### PR DESCRIPTION
**Pull request by SimonDarksideJ**
Draft pass to update the InputSystem to use the "Logical" InputAction instead of the old "Physical" InputType (which has been renamed to DeviceInputType for clarity)

* Moved Device specific device classes / enums etc to new "Definitions/Devices" folder
* Functions updated to use new type
* Initialisers / validators updated as best guess

# TODO - @StephenHodgson

- [ ] Review InteractionDefiniton constructor
- [ ] Review (TODOs) where references to InputAction are made
- [ ] Review "SupportsCapability", if this is used in InputSystem, shouldn't it use Logical, NOT physical.

### CODE REVIEW

File: Assets/MixedRealityToolkit/InputSystem/MixedRealityInputManager.cs [hunk 1]
```diff
@@ -558,12 +558,12 @@ public void RaisePointerDown(IMixedRealityPointer pointer, Handedness handedness
...
         /// <inheritdoc />
-        public void RaisePointerDown(IMixedRealityPointer pointer, Handedness handedness, InputType inputType)
+        public void RaisePointerDown(IMixedRealityPointer pointer, Handedness handedness, InputAction inputAction)
         {
             // Create input event
-            inputClickEventData.Initialize(pointer.InputSourceParent, handedness, inputType);
+            inputClickEventData.Initialize(pointer.InputSourceParent, handedness, inputAction);
 
-            if (inputType == InputType.Select)
+            if (inputAction.Description == "Select") // TODO - Unsure how the new InputActions are Referred to?
```
Comment (StephenHodgson) [inline Assets/MixedRealityToolkit/InputSystem/MixedRealityInputManager.cs]: Not sure string comparisons is the best way to do this.

Comment (SimonDarksideJ) [reply Assets/MixedRealityToolkit/InputSystem/MixedRealityInputManager.cs]: Lol, no, that was just one attempt, I used ID in another class, I'll fix that

**Comment by SimonDarksideJ** [comment]
Well, you need to at least define the default "Internal" InputActions like we had in the Enum.
But I do prefer this approach as it allows to create their own actions, which the inputsystem will just pass along.
I just need to be able to use them in a dropdown in the editor for the "Mapping"

**Comment by StephenHodgson** [comment]
That was my thoughts as well.

File: Assets/MixedRealityToolkit/_Core/Definitions/InputSystem/MixedRealityInputActionsProfile.cs [hunk 2]
```diff
@@ -11,6 +14,58 @@ namespace Microsoft.MixedReality.Toolkit.Internal.Definitions.InputSystem
...
+        [Tooltip("Array of MRTK Input Actions and their ID")]
+        private string[] inputActions = new string[4] {"Select", "Menu", "Pointer", "Grip"}; // TODO - Need to populate all the MRTK Default Actions
 
+        public string[] InputActions { get { return inputActions; } }
     }
+
+#if UNITY_EDITOR
+    //TODO - Move this sucker elsewhere :D
+    [CustomEditor(typeof(MixedRealityInputActionsProfile))]
+    public class MixedRealityInputActionsProfileInspector : Editor
```
Comment (StephenHodgson) [inline Assets/MixedRealityToolkit/_Core/Definitions/InputSystem/MixedRealityInputActionsProfile.cs]: not in it's own class via inspectors?

Comment (SimonDarksideJ) [reply Assets/MixedRealityToolkit/_Core/Definitions/InputSystem/MixedRealityInputActionsProfile.cs]: lol, see comment. i ran out of time.

File: Assets/MixedRealityToolkit/_Core/Definitions/InteractionDefinition.cs [hunk 3]
```diff
@@ -12,11 +12,11 @@ namespace Microsoft.MixedReality.Toolkit.Internal.Definitions.InputSystem
     /// </summary>
     public struct InteractionDefinition
     {
-        public InteractionDefinition(uint id, AxisType axisType, InputType inputType) : this()
+        public InteractionDefinition(uint id, AxisType axisType, Devices.DeviceInputType inputType) : this()
         {
             Id = id;
             AxisType = axisType;
-            InputType = inputType;
+            InputType = inputType;  // TODO - SHould this in the constructor now?  As it's input, should the InputAction be here.
```
Comment (StephenHodgson) [inline Assets/MixedRealityToolkit/_Core/Definitions/InteractionDefinition.cs]: Yes, I don't see why not after the InputAction changes I've made.

**Pull request merged**